### PR TITLE
allow methodless macros #22098 (fcards work)

### PR DESCRIPTION
**Pull request by jw3126**
Credit goes to @fcard, see #22098. I am just the typesetter.

### CODE REVIEW

**Review by vtjnash — APPROVED** [review]

File: src/julia-parser.scm [hunk 1]
```diff
@@ -1406,11 +1406,11 @@
        ((function macro)
         (let* ((paren (eqv? (require-token s) #\())
                (sig   (parse-def s (not (eq? word 'macro)))))
-          (if (and (eq? word 'function) (not paren) (symbol-or-interpolate? sig))
+          (if (and (not paren) (symbol-or-interpolate? sig))
               (begin (if (not (eq? (require-token s) 'end))
-                         (error (string "expected \"end\" in definition of function \"" sig "\"")))
+                          (error (string "expected \"end\" in definition of " word " \"" sig "\"")))
```
Comment (JeffBezanson) [inline src/julia-parser.scm]: indentation

File: src/julia-syntax.scm [hunk 2]
```diff
@@ -1221,6 +1221,8 @@
                                           v))
                                     anames))
                        ,@(cddr e)))))
+        ((and (length= e 2) (symbol? (cadr e)))
+                  (expand-forms `(function ,(symbol (string #\@ (cadr e))))))
```
Comment (JeffBezanson) [inline src/julia-syntax.scm]: indentation

File: test/core.jl [hunk 3]
```diff
@@ -5649,3 +5649,9 @@ function hh6614()
     x, y
 end
 @test hh6614() == (1, 2)
+
+# issue 22098
+macro m22098 end
+handle_on_m22098 = getfield(@__MODULE__, Symbol("@m22098"))
+methods_m22098 = methods(handle_on_m22098)
+@test isempty(methods_m22098)
```
Comment (JeffBezanson) [inline test/core.jl]: No need for the intermediate variables here.

Comment (jw3126) [reply test/core.jl]: I find it more readable like this. What is the benefit of spelling it 
```julia
@test isempty(methods(getfield(@__MODULE__, Symbol("@m22098"))))
```
?

Comment (jw3126) [reply test/core.jl]: I agree on removing the `methods_m22098` variable though.

File: src/julia-parser.scm [hunk 4]
```diff
@@ -1406,11 +1406,11 @@
        ((function macro)
         (let* ((paren (eqv? (require-token s) #\())
                (sig   (parse-def s (not (eq? word 'macro)))))
-          (if (and (eq? word 'function) (not paren) (symbol-or-interpolate? sig))
+          (if (and (not paren) (symbol-or-interpolate? sig))
               (begin (if (not (eq? (require-token s) 'end))
-                         (error (string "expected \"end\" in definition of function \"" sig "\"")))
+                       (error (string "expected \"end\" in definition of " word " \"" sig "\"")))
```
Comment (JeffBezanson) [inline src/julia-parser.scm]: Sorry to hound you about this but the indentation is still wrong.

**Comment by JeffBezanson** [comment]
Thanks for putting up a PR for this, @jw3126 .

File: src/julia-parser.scm (hunk 4 above)
Comment (jw3126) [reply src/julia-parser.scm]: No problem can you explain how the indentation works? This is what vim suggests and also what I thought was right. It is one level deeper then the above `(if` so indent by two spaces?

Comment (jw3126) [reply src/julia-parser.scm]: Ok it needs to align with the `(not`, now I get it.

**Pull request merged**